### PR TITLE
Persist filters in URL

### PR DESCRIPTION
**Pull request by jorntx**
A really basic (but hard) first go at persisting filters in the URL

### CODE REVIEW

File: src/components/search/wc-facet-checkbox.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ class checkboxComponent extends HTMLElement {
 						new CustomEvent('filter-update', {
 							detail: {
 								filter: `fq=${this.key}:"${this.sendValue}"`,
+								rawFilter: `${this.key}:"${this.sendValue}"`,
```
Comment (jesperlauridsen) [inline src/components/search/wc-facet-checkbox.ts:26]: filter and rawFilter are kinda they same. Would it be simpler to make filter the same value as rawFilter now, and simply add 'fq=' where it's needed when used to apply the filters to the URL? I'm not sure, just asking questions here...

File: src/components/search/Facets.vue [hunk 2]
```diff
@@ -82,11 +80,33 @@ export default defineComponent({
...
+				} else if (Array.isArray(routeQueries.fq)) {
+					routeQueries.fq.push(newFilter);
+				} else {
+					//This will only trigger if someone manipulates the url manually
+					routeQueries.fq = [routeQueries.fq, newFilter];
+				}
 				this.searchResultStore.addFilter(e.detail.filter);
 			} else {
+				const filterToRemove = encodeURIComponent(e.detail.rawFilter);
+				routeQueries.fq = routeQueries.fq.filter((item: string) => item !== filterToRemove);
```
Comment (jesperlauridsen) [inline src/components/search/Facets.vue:106]: This is some very nice coding. I had to read that a few times before i was sure what this did exactly :D

File: src/components/search/wc-facet-checkbox.ts (hunk 1 above)
Comment (jorntx) [reply src/components/search/wc-facet-checkbox.ts:26]: Valid point and I had the exact same idea but eventually got too lazy to do the refactor, so added the param.

I think your call is good and we should take the time to rectify - so I will make a separate ticket for this.

File: src/components/search/Facets.vue (hunk 2 above)
Comment (jorntx) [reply src/components/search/Facets.vue:106]: This is ES6 at it's finest but please do speak up if it gets too compact.

File: src/components/search/wc-facet-checkbox.ts (hunk 1 above)
Comment (jorntx) [reply src/components/search/wc-facet-checkbox.ts:26]: Ticket made - DISC-349

**Review by jesperlauridsen — APPROVED** [review]

**Comment by jesperlauridsen** [comment]
Seems good - I'll approve and merge!

**Pull request merged**